pkg/catalog/catalog.go

```diff
@@ -232,3 +300,14 @@ func (c *Catalog) loadProjects() ([]*v1alpha1.Project, error) {
 
 	return projects, nil
 }
+
+type catalogKey struct{}
+
+func ToContext(ctx context.Context, catalog *Catalog) context.Context {
+	return context.WithValue(ctx, catalogKey{}, catalog)
+}
+
+func FromContext(ctx context.Context) *Catalog {
+	catalog, _ := ctx.Value(catalogKey{}).(*Catalog)
```

Comment:
Just a general comment that we are validating for nil nowhere when using `FromContext`, so we'll be getting nil refs panics. Maybe check it here and panic with a clearer error message, rather than letting it panic in more obscure places downstream?

Reply:
Philosophically I am against doing this, because then calling code can't check for a catalog and then, take an action with regards to if there is a catalog or not. Calling this function will panic.

In practice, its ok for now given that we never call fromContext unless we expect the catalog to be there, so I'll do it but maybe we will revert in future if we get a conditional use-case.